framework/wazuh/utils.py

```diff
@@ -764,9 +764,12 @@ def _parse_legacy_filters(self):
         """
         Parses legacy filters.
         """
-        legacy_filters_as_list = {name: value.split(',') if isinstance(value, unicode) or isinstance(value,str) else value for name, value in self.legacy_filters.items()}
-        self.query_filters += [{'value': None if subvalue == "null" else subvalue, 'field': '{}${}'.format(name,i), 'operator': '=', 'separator': 'OR' if len(value) > 1 else 'AND', 'level': 0}
-                               for name, value in legacy_filters_as_list.items() for subvalue,i in zip(value, range(len(value))) if not self._pass_filter(subvalue)]
+        legacy_filters_as_list = {name: value.split(',') if isinstance(value,str) else value for name, value in self.legacy_filters.items()}
+        self.query_filters += [{'value': None if subvalue == "null" else subvalue, 'field': '{}${}'.format(name,i),
```

Comment:
We should consider the case when `value` is a number which doesn't have the len property. Let's change the unit tests to detect this potential error and ensure the correctness of the implementation :)

Reply:
You were right, the filter failed when a non-string value was used as a filter value. Fixed in https://github.com/wazuh/wazuh/pull/2869/commits/280b61021171628836518d6cbd7039fd07e8dea0